### PR TITLE
Show block handles only on mouse hover

### PR DESCRIPTION
**Pull request by yusufkinatas**
## 🌟 What is the purpose of this PR?

- Hides the `BlockHandle` by default, and display it only when the mouse cursor hovers over a block.
- Changes a spelling to American English within a comment

## 🔗 Related links
[Asana task](https://app.asana.com/0/1200211978612931/1202700524579036/f) (internal)

## ❓ How to test this?

1.  Checkout the branch / view the deployment
1.  Try hovering on a Block inside a page
1.  Confirm that `BlockHandle` is visible on hover.

## 📹 Demo

https://user-images.githubusercontent.com/39553853/183638037-1045618b-1127-412b-ab4b-c0ea40881a13.mov

### CODE REVIEW

**Comment by nonparibus** [comment]
The video demo looks perfect. Will await someone else for proper functional user acceptance testing (UAT) and code review, but good job on getting your first PR out, @yusufkinatas! :)

**Review by luisbettencourt — CHANGES_REQUESTED** [review]
I suggest moving this logic to the `BlockHandle` component. We can also track if the menu is open there to tackle https://hashintel.slack.com/archives/C022217GAHF/p1660049753582549.

**Comment by yusufkinatas** [comment]
> I suggest moving this logic to the `BlockHandle` component. We can also track if the menu is open there to tackle https://hashintel.slack.com/archives/C022217GAHF/p1660049753582549. Let me know if you think otherwise or if you have any questions.

This is adressed at https://github.com/hashintel/hash/pull/907/commits/53da573ba6033a8aab4030f5ec641eeb593aec9b

File: packages/hash/frontend/src/blocks/page/BlockView.tsx [hunk 1]
```diff
@@ -237,6 +250,8 @@ export class BlockView implements NodeView<Schema> {
     this.renderPortal(null, this.selectContainer);
     this.dom.remove();
     document.removeEventListener("dragend", this.onDragEnd);
+    document.removeEventListener("mouseenter", this.onMouseEnter);
+    document.removeEventListener("mouseleave", this.onMouseLeave);
```
Comment (luisbettencourt) [inline packages/hash/frontend/src/blocks/page/BlockView.tsx:254]:
```suggestion
    this.dom.removeEventListener("mouseenter", this.onMouseEnter);
    this.dom.removeEventListener("mouseleave", this.onMouseLeave);
```

**Review by luisbettencourt — CHANGES_REQUESTED** [review]

**Review by luisbettencourt — APPROVED** [review]
Looks good to me, great work @yusufkinatas !

**Comment by luisbettencourt** [comment]
I don't think it should stop merging this PR, but one thing that I notice because of this change is that when we load an empty page it can be hard to know if the page has loaded or if there are anything to edit at all:

demo

What do you think about this @nonparibus ?

**Comment by nonparibus** [comment]
> I don't think it should stop merging this PR, but one thing that I notice because of this change is that when we load an empty page it can be hard to know if the page has loaded or if there is anything to edit at all:
> 
> demo [ demo ](https://user-images.githubusercontent.com/37453800/183686467-cabd4f51-237c-4131-9ca6-dfc2f606a45f.gif) [ ](https://user-images.githubusercontent.com/37453800/183686467-cabd4f51-237c-4131-9ca6-dfc2f606a45f.gif)
> 
> What do you think about this @nonparibus ?

We should:
1. Add skeleton loaders for headers, and blocks on pages
1. Have a 'blank page' state
1. Add a prompt on each new line

We have a ticket to create designs for 1 ([internal link](https://app.asana.com/0/0/1201418039648674/f)), and finished designs for 2 ([internal link](https://www.figma.com/file/gydVGka9FjNEg9E2STwhi2/HASH-App?node-id=1564%3A102478)) and 3 ([internal link](https://www.figma.com/file/gydVGka9FjNEg9E2STwhi2/HASH-App?node-id=1564%3A102613)) (but no implementation/dev ticket yet).

**Pull request merged**